FHIRServicesExternalModule.php

```diff
@@ -2428,19 +2428,37 @@ function getEConsentFHIRBundle($args){
         $firstName = $args['firstName'];
         $lastName = $args['lastName'];
         $patientId = $args['patientId'];
+        $birthDate = $args['birthDate'];
 
-        $patient = $this->createResource('Patient', [
-            'id' => $patientId,
-            'name' => [
+        $patientObj = [
+            'id' => $patientId
+        ];
+
+        $allow_cleaned_consent_bundle = $this->getProjectSetting('allow-consent-bundle-cleanup');
+
+        // Set attributes only if not null
+        if(is_null($firstName) || is_null($lastName)){
+            if(!$allow_cleaned_consent_bundle){
+                throw new \Exception('eConsent FHIR name attribute is null.');
+            }
+        } else {
+            $patientObj['name'] = [
                 [
-                    'given' => [
-                        $firstName
-                    ],
+                    'given' => [ $firstName ],
                     'family' => $lastName
                 ]
-            ],
-            'birthDate' => $args['birthDate']
-        ]);
+            ];
+        }
+
+        if(is_null($birthDate)){
```

Comment:
Does `is_null()` work is all cases, or would it be more appropriate to use `empty()` to support empty string values as well in all three places?

Reply:
Good point. Technically, it is allowed to set an empty string according to the FHIR specs as far as I understand. On the other hand, empty strings do not make sense semantically.
My objective was to prevent technical errors during the FHIR validation from the server, so I used `is_null` initially.
Using `empty()` changes the initial behaviour of the plugin towards empty strings such that empty strings are now not accepted, but have been valid up till now. Since I did not want to change such behaviour and wanted to keep the "default" behaviour in this PR, `is_null()` was used.
The final decision is up to you. :)

Reply:
I would like to leave it up to you.  I am not planning on testing your changes, as I'm not really supposed to be spending any  time on this module since it does not have a support budget.  If you've tested it and `is_null()` works better, I trust you.

Reply:
Alright, I'll think about it for a while... Maybe I will figure out some situations where the design choice matters.

Reply:
I think we should stick to `is_null()` instead of using `empty()` for the following reasons:
- Default behaviour: using empty() would change the default behaviour to prior plugin's behaviour
- Conflicting situation with only lastname (discussed below): In situations where only last name is provided, empty() would cause problems with missing first names. One could differentiate between first name and last name but this would be an unpredictable expected behavior for external users.

Reply:
Sounds good to me.  Resolving...